test/runtests.jl

```diff
@@ -318,6 +318,14 @@ x = XInt(10)
 @test JSON3.read("{\"a\": 1, \"b\": 2}", NamedTuple{(:a, :b), Tuple{Float64, Float64}}) == (a=1.0, b=2.0)
 @test JSON3.write((a=1.0, b=2.0)) == "{\"a\":1.0,\"b\":2.0}"
 
+# Test that writing DictType works even when length(pairs(dict_type)) isn't
+# available. Issue #37.
+struct MyDictType end
```

Comment:
You need to move this definition to the top of the file, above the `@testset` definition for 1.0 to pass tests.

Reply:
Sorry about that. I thought I had checked that CI was passing, but apparently I ignored it entirely.